src/NuGet.Core/NuGet.ProjectModel/PackageSpecWriter.cs

```diff
@@ -67,6 +67,45 @@ public static void Write(PackageSpec packageSpec, IObjectWriter writer)
         }
 
         /// <summary>
+        /// Writes a PackageSpec with only the restore info that's important for hashing/no-op to an <c>NuGet.Common.IObjectWriter</c> instance. 
+        /// </summary>
+        /// <param name="packageSpec">A <c>PackageSpec</c> instance.</param>
+        /// <param name="writer">An <c>NuGet.Common.IObjectWriter</c> instance.</param>
+        public static void WriteHashingPackageSpec(PackageSpec packageSpec, IObjectWriter writer)
+        {
+            if (packageSpec == null)
+            {
+                throw new ArgumentNullException(nameof(packageSpec));
+            }
+
+            if (writer == null)
+            {
+                throw new ArgumentNullException(nameof(writer));
+            }
+
+            SetValue(writer, "title", packageSpec.Title);
+
+            if (!packageSpec.IsDefaultVersion)
+            {
+                SetValue(writer, "version", packageSpec.Version?.ToFullString());
+            }
+            // We don't need details like Description, Authors, Copyright, Language
+            // We do not need pack related thing, PackInclude & packOptions
+            SetArrayValue(writer, "contentFiles", packageSpec.ContentFiles);
```

Comment:
I don't think we use this, there are a lot of properties in PackageSpec that were used in the past for packing the project.json file in the same way a nuspec is passed. They aren't used in restore so you should ignore them during hashing.